### DISCUSSION TITLE
No funciona con tienda configurada en dolares

### DISCUSSION THREAD

**Discussion by linkini** [whooohq/whq-woocommerce-chilexpress-shipping#222] [Support]
Tengo una tienda que trabaja con los productos en dólares. Quisiera saber qué archivo modificar de forma "segura" (entre comillas) para hacer una conversión (sea por algún servicio con API o yo fijando un valor dólar fijo).
Agradecería respuesta a la brevedad.

Saludos

**Comment by EstebanForge** [comment]
@linkini buen día.

Lo que solicitas está fuera de lo que pretende resolver este plugin.

Puedes probar con https://wordpress.org/plugins/woocommerce-chilean-peso-currency/
No lo he probado personalmente. Es solo una sugerencia.

Esteban

**Reply by linkini** [reply]
El plugin podría generar conflicto porque también trae las regiones de Chile.
¿Cuál es el archivo PHP que devuelve el valor en CLP? Puedo modificar las lineas correspondientes cada vez que haya actualización del plugin, no habría problema.

Muchas gracias!

**Reply by EstebanForge** [reply]
Puedes descargar el plugin aquel, y luego editar el archivo woocommerce-chilean-peso.php para cambiar la linea 156, que dice:

```
add_filter('woocommerce_states', 'custom_woocommerce_states');
```

Para comentarla. Déjala así:
```
//add_filter('woocommerce_states', 'custom_woocommerce_states');
```

Eso evitará que el plugin aquel agrege las regiones a WC (que no funcionan con Chilexpress), y podrás beneficiarte de la conversión automática de CLP->USD que hace aquel plugin.

No lo he probado, pero debería funcionar.

Esteban

**Reply by linkini** [reply]
Es que yo no busco pasar dolares a pesos o pesos a dolares hablando en general de la tienda. Yo solo necesito que vuestro tarificador entregue valor en dolares, solo consulto por el nombre de archivo donde despliega el valor, para ahi yo hacer una reprogramación, asumiendo el riesgo que implica en futuras actualizaciones (lo tengo super claro). La tienda funciona con dólares como moneda oficial y solo procesará pagos en paypal, mi única piedra de tope es el valor de envío en CLP. Es lo unico que quiero tocar. Lamentablemente es el cliente el que me amarra a esto. Favor indicarme el nombre del archivo que entrega el valor de envío en checkout, eso es todo.

**Reply by EstebanForge** [reply]
La versión 1.4.35 recién subida, tiene un filtro nuevo: whq_wcchp_rates_cost
Ese filtro, junto con uno previamente existente, whq_wcchp_tarification_rates, te permiten modificar los datos que llegan desde la API de Chilexpress antes de ser mostrada.

Con whq_wcchp_tarification_rates filtras y modificas los "rates" que son generados y obtenidos luego de llamada a la API de Chilexpress.

Con whq_wcchp_rates_cost filtras y modificas el costo de envío uno por uno al momento previo de ser agregados a WC por este método de envío.

Son filtros.
No necesitas modificar el plugin para poder alterarlo. Simplemente aplica tus filtros (con esos nombres que especifiqué, dependiendo de lo que quieras hacer) y modificas al vuelo lo que quieres que salga https://developer.wordpress.org/plugins/hooks/filters/

En esos puntos puedes tomar el valor en CLP y convertirlo a otra moneda antes de que el método de envío lo agregue a WC.

Espero te sirva.

Esteban
